Client/WebFilters/ShieldStats/Adblock/AdBlockStats.swift

```diff
@@ -62,11 +65,17 @@ class AdBlockStats: LocalAdblockResourceProtocol {
         let filePaths = enumerator?.allObjects as? [URL]
         let datFileUrls = filePaths?.filter { $0.pathExtension == "dat" }
         
-        datFileUrls?.forEach {
-            let fileName = $0.deletingPathExtension().lastPathComponent
-            
-            guard let data = fm.contents(atPath: $0.path) else { return }
-            setDataFile(data: data, id: fileName)
+        downloadTask?.cancel()
```

Comment:
You're cancelling the task but I don't see any check for cancellation status within the Task below. 

Add `try Task.checkCancellation()` inside the `asyncConcurrentForEach` closure.  You may also want to check for cancellation inside `setDataFile`